Sources/XCTest/XCTestCase.swift

```diff
@@ -72,25 +76,33 @@ extension XCTestCase {
     }
 
     internal static func invokeTests(tests: [(String, XCTestCase throws -> Void)]) {
+        let observationCenter = XCTestObservationCenter.sharedTestObservationCenter()
+
         var totalDuration = 0.0
         var totalFailures = 0
         var unexpectedFailures = 0
         let overallDuration = measureTimeExecutingBlock {
             for (name, test) in tests {
-                let testCase = self.init()
-                let fullName = "\(testCase.dynamicType).\(name)"
+                let testCase = self.init(methodName: name)
 
                 var failures = [XCTFailure]()
                 XCTFailureHandler = { failure in
+                    observationCenter.testCase(testCase,
+                                               didFailWithDescription: "\(failure.failureDescription) - \(failure.message)",
+                                               inFile: String(failure.file),
+                                               atLine: failure.line)
```

Comment:
The description we pass to the observers is different from the description that's printed by `XCTFailure.emit()`. Is this intentional? It seems like it opens us up to some maintenance dilemmas down the line... for example, @mike-ferris-apple [mentioned looking into removing trailing whitespace from failure messages](https://github.com/apple/swift-corelibs-xctest/pull/71#issuecomment-196318176). This code would necessitate us to modify two different call paths.

Could we have `XCTFailure.emit()` return a string, which we then pass to observers? Or is the description passed to observers supposed to be different here? I'll try checking to see what Apple XCTest does myself to make sure.

Of course, this could be something we take care of later. Let me know if you'd rather we address this in later commits.